src/Package/Impl/DataInspect/Viewers/FunctionViewer.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Threading.Tasks;
+using EnvDTE;
+using Microsoft.Common.Core;
+using Microsoft.R.Core.Formatting;
+using Microsoft.R.Debugger;
+using Microsoft.R.Editor.Settings;
+using Microsoft.R.Editor.Signatures;
+using Microsoft.R.Host.Client;
+using Microsoft.VisualStudio.R.Package.Repl;
+using Microsoft.VisualStudio.R.Package.Shell;
+using static System.FormattableString;
+
+namespace Microsoft.VisualStudio.R.Package.DataInspect.Viewers {
+    [Export(typeof(IObjectDetailsViewer))]
+    internal sealed class FunctionViewer : IObjectDetailsViewer {
+        private readonly IRSessionProvider _sessionProvider;
+        private readonly IDebugObjectEvaluator _evaluator;
+
+        [ImportingConstructor]
+        public FunctionViewer(IRSessionProvider sessionProvider, IDebugObjectEvaluator evaluator) {
+            _sessionProvider = sessionProvider;
+            _evaluator = evaluator;
+        }
+
+        #region IObjectDetailsViewer
+        public ViewerCapabilities Capabilities => ViewerCapabilities.Function;
+
+        public bool CanView(DebugValueEvaluationResult evaluation) {
+            return evaluation != null && evaluation.Classes.Count == 1 && evaluation.Classes[0].EqualsOrdinal("function");
+        }
+
+        public async Task ViewAsync(string expression, string title) {
+            var evaluation = await _evaluator.EvaluateAsync(expression, DebugEvaluationResultFields.Expression) as DebugValueEvaluationResult;
+            if(evaluation == null) {
```

Comment:
Do we want to silently ignore eval errors? It might be worth checking for `DebugErrorEvaluationResult` here, and displaying the error to the user.